server/modules/survey/manager/surveyManager.js

```diff
@@ -93,12 +92,22 @@ const fetchSurveyById = async (surveyId, draft = false, validate = false, client
   return assocSurveyInfo(info)
 }
 
-const fetchSurveyAndNodeDefsBySurveyId = async (id, draft = false, advanced = false, validate = false, client = db) => {
-  const survey = await fetchSurveyById(id, draft, validate, client)
-  const nodeDefs = await NodeDefManager.fetchNodeDefsBySurveyId(id, draft, advanced, validate, client)
+const fetchSurveyAndNodeDefsBySurveyId = async (surveyId, draft = false, advanced = false, validate = false, client = db) => {
+  const survey = await fetchSurveyById(surveyId, draft, validate, client)
+  const nodeDefs = await NodeDefManager.fetchNodeDefsBySurveyId(surveyId, draft, advanced, validate, client)
   return Survey.assocNodeDefs(nodeDefs)(survey)
 }
 
+const fetchSurveyAndNodeDefsAndRefDataBySurveyId = async (surveyId, draft = false, advanced = false, validate = false, client = db) => {
+  const survey = await fetchSurveyAndNodeDefsBySurveyId(surveyId, draft, advanced, validate, client)
+  const [categoryIndexRS, taxonomyIndexRS] = await Promise.all([
+    CategoryRepository.fetchIndex(surveyId, draft, client),
+    TaxonomyRepository.fetchIndex(surveyId, draft, client)
+  ])
+
+  return Survey.assocRefData(categoryIndexRS, taxonomyIndexRS)(survey)
+}
+
```

Comment:
Why not split the two functions above into two and three separate functions?

Reply:
Because the SurveyManger responsibility is to compose the objects to be consumed by outside callers.